corehq/apps/reports/standard/sms.py

```diff
@@ -170,7 +167,18 @@ def _fmt_contact_link(self, recipient_id, doc_info, extra_text=None):
         username = username or "-"
         contact_type = contact_type or _("Unknown")
         if url:
-            ret = self.table_cell(username, '<a target="_blank" href="%s">%s</a>' % (url, username))
+            # We need to store the contact type and the recipient ID in the contact link element, because
+            # MessageLogReport can only use the link, not the whole table cell dictionary, and so it won't have
+            # table_cell['raw'] available for export. export_table will read contacttype and recipientid
+            # attributes to populate those columns in the export spreadsheet.
+            ret = self.table_cell(
+                username,
+                '<a target="_blank" href="{url}" contacttype="{type}" recipientid="{id}">{username}</a>'.format(
+                    username=username,
+                    url=url,
+                    type=contact_type,
+                    id=recipient_id or ''
+                ))
```

Comment:
I think we should try to avoid adding the values as html attributes. Also I don't think you'll have access to `cell['raw']` from `export_rows` since it just gets its data from `rows` which doesn't use table cells anymore, unless I'm missing something?

I think one way to solve this problem is just to have a method or property separate from `rows` and `export_table` that generates all of the data, including the extra columns needed for export.  Then have `rows` exclude the extra columns.

How come we removed the use of table cells, was that to reduce on memory usage?